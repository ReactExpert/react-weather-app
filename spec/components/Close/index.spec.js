import React from 'react';
import Close from '../../../src/components/Close';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('Close', () => {
  it('matches snapshot', () => {
    const component = mount(<Close onCloseClick={() => { }} />);

    expect(component).toMatchSnapshot();
  });

  describe('animate', () => {
    it('adds class animate-in to button', (done) => {
      const component = mount(<Close onCloseClick={() => { }} />);

      component.instance().animate();

      setTimeout(() => {
        expect(component.instance().button.current.className).toContain('animate-in');
        done();
      }, 100);
    });
  });

  describe('hide', () => {
    it('removes class animate-in from button', (done) => {
      const component = mount(<Close onCloseClick={() => { }} />);

      component.instance().hide();

      setTimeout(() => {
        expect(component.instance().button.current.className).not.toContain('animate-in');
        done();
      }, 100);
    });
  });
});
