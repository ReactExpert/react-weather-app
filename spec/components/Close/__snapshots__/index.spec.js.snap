// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close matches snapshot 1`] = `
<Close
  onCloseClick={[Function]}
>
  <button
    aria-label="Close view"
    className="button-close"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="#fff"
      height="24"
      stroke="#000"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6L6 18M6 6l12 12"
      />
    </svg>
  </button>
</Close>
`;
